python-backend/app/api/mines/variances/resources/variance.py

```diff
@@ -0,0 +1,75 @@
+from datetime import datetime
+from flask_restplus import Resource, reqparse
+from sqlalchemy.exc import DBAPIError
+from werkzeug.exceptions import BadRequest, NotFound
+
+from ..models.variance import Variance
+from app.extensions import api
+from ....utils.access_decorators import requires_role_mine_view, requires_role_mine_create
+from ....utils.resources_mixins import UserMixin, ErrorMixin
+
+
+class VarianceResource(Resource, UserMixin, ErrorMixin):
+    parser = reqparse.RequestParser()
+    parser.add_argument('compliance_article_id',
+                        type=int,
+                        help='ID representing the MA or HSRCM code to which this variance relates.')
+    parser.add_argument('note',
+                        type=str,
+                        help='A note to include on the variance. Limited to 300 characters.')
+    parser.add_argument('issue_date',
+                        type=lambda x: datetime.strptime(x, '%Y-%m-%d') if x else None,
+                        help='The date on which the variance was issued.')
+    parser.add_argument('received_date',
+                        type=lambda x: datetime.strptime(x, '%Y-%m-%d') if x else None,
+                        help='The date on which the variance was received.')
+    parser.add_argument('expiry_date',
+                        type=lambda x: datetime.strptime(x, '%Y-%m-%d') if x else None,
+                        help='The date on which the variance expires.')
```

Comment:
Can you add `store_missing=False` to each of these and `trim=True` to note? 

Changing the `compliance_article_id` to .get() doesn't change anything else.

Reply:
I can add those for consistency, but it's a code table, so nothing is being updated via the API. It's just a static data endpoint.

Reply:
It's the variance resource. business dates and notes. Post is in this, and put will be coming next sprint.

Reply:
Oh jeez. Yeah, you're right. Thanks